### ISSUE TITLE
Test Explorer does not convert failed test to passing after save

### ISSUE THREAD

**Issue by JustinGrote** [jest-community/vscode-jest#757]
Issue Type: <b>Bug</b>

![Capture](https://user-images.githubusercontent.com/15258962/130707047-b73ffd2d-cd95-411f-8c05-6032e144974d.gif)

Extension version: 4.1.1
VS Code version: Code - Insiders 1.60.0-insider (44568e5947efa4811419e529e80fb14b7554cf61, 2021-08-23T10:43:01.389Z)
OS version: Windows_NT x64 10.0.22000
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 7 4700U with Radeon Graphics          (8 x 1996)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.36GB (3.99GB free)|
|Process Argv|--folder-uri file:///c%3A/Users/JGrote/Projects/vscode-adapter --crash-reporter-id 4453efaa-0821-4555-88e2-3de8bad9109e|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
pythonvspyt602:30291494
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt639:30291487
pythontb:30258533
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
pythonvsuse255:30319630
vscod805:30301674
pythonvspyt200:30323110
vsccppwtct:30312693
pythonvssor306:30340298
bridge0708:30335490
pygetstartedt2:30353727
dockerwalkthrucf:30348714
bridge0723:30353136
pythonrunftest32cf:30353182
pythonf5test824:30353183
javagetstartedt:30350119
pythonvspyt187cf:30356035
pydsgst2:30356659

```

</details>

**Comment by JustinGrote** [comment]
Potential dupe of https://github.com/jest-community/vscode-jest/issues/734, unconfirmed

**Comment by JustinGrote** [comment]
This appears to be because if the code matches the last commit, the test will not be run. May need to force override this behavior in jest for the individual file.

Output Window after saving back to "true"
```
> Test run finished at 8/24/2021, 5:31:22 PM <

No tests found related to files changed since last commit.
```

**Comment by connectdotz** [comment]
I assume you are in watch mode? It's a known issue about how jest watcher worked.  You can switch to the interactive mode (see autoRun), which triggers run by editor save, regardless of git status.

**Comment by raph90** [comment]
@connectdotz can you please clarify on this? I'm not sure if the problem is in my launch.json or elsewhere. However this is currently making the extension unusable... 
How would I switch to the interactive mode? Is that done via the extension settings or with the launch.json?

**Comment by alexamy** [comment]
**TL;DR**
Set this editor setting to trigger test run on each save:
```javascript
"jest.autoRun": { "watch": false, "onSave": "test-src-file", "onStartup": ["all-tests"] }
```

**Comment by jmagaram** [comment]
I'm having a similar issue. I see the error like `Expected 100, Received 1` but the actual javascript test file expects the number `2`. The only way I can fix this is to force the tests to run again by either saving the javascript file or switching to manual mode. In my case I'm using ReScript and the test files are generated automatically so I can't easily open them up and save them. There clearly is a problem here. Even if I set the tests in watch mode nothing changes when new test files are added or I edit my ReScript and an updated test file is generated. At this point I have to use manual mode.

**Comment by connectdotz** [comment]
@jmagaram 

> This is from a create-react-app with typescript. I added ReScript. If I run "test": "react-scripts test", in watch mode it works fine. 

If `yarn test` or `npm test` works from a terminal, try setting your `jest.jestCommandLine` to that (append additional `--` for npm command). In your use case, I think watch mode is probably better than manual/interactive mode since the test files are generated behind the scene...

If the above did not work, then maybe provide a sample repo so we can further investigate...
